### PR TITLE
Added markdown documentation for ledger events

### LINKED ISSUE #2739: Document the ledger events

### PR DESCRIPTION
**Pull request by JaredCorduan**
This provides markdown documentation for the ledger events.

closes  #2739

### CODE REVIEW

File: docs/LedgerEvents.md [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+## Overview
+
+The main API function provided by the ledger, `applyBlockOpts`, can be configured to return ledger events by
+setting the event policy to `EPReturn` in the `asoEvents` field of `ApplySTSOpts`.
+The events come at the cost of performance, but are useful for understanding changes to the ledger state
+after each block is applied. Note that there are no guarantees about the ordering of the events
+for any given block.
+
+Note thet when we say that an event only occurs on an epoch boundary, we mean that it only occurs on the first
+block of a new ear.
```
Comment (lehins) [inline docs/LedgerEvents.md]: `ear`? :smiley: Maybe `epoch`?

File: docs/LedgerEvents.md [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+
+## List of Events
+
+### `NewEpoch epoch`
+
+This event signals the fact that the given block is the first block of a new epoch, namely epoch `epoch`.
```
Comment (lehins) [inline docs/LedgerEvents.md]: Too many epochs :stuck_out_tongue_winking_eye: 
```suggestion
This event signals the fact that the given block is the first block of a new epoch, namely `epoch`.
```

File: docs/LedgerEvents.md [hunk 3]
```diff
@@ -0,0 +1,124 @@
...
+(a change in stake pool parameters).
+The field `poolID` is the stake pool ID.
+
+### `SuccessfulPlutusScriptsEvent debugInformation`
+This event happens for every collection of Plutus scripts in a given transaction
+which successfully execute.
+The field `debugInformation` is a list of `PlutusDebug`, which is a record containing
+all the information needed to re-run a Plutus script.
+
+### `FailedPlutusScriptsEvent (scriptFailuresToPlutusDebug fs`
```
Comment (lehins) [inline docs/LedgerEvents.md]:
```suggestion
### `FailedPlutusScriptsEvent (scriptFailuresToPlutusDebug fs)`
```

**Review by lehins — APPROVED** [review]
Looks good. More documentation is always better

File: docs/LedgerEvents.md (hunk 3 above)
Comment (JaredCorduan) [reply docs/LedgerEvents.md]: oops, that's a copy paste error, should have been `FailedPlutusScriptsEvent debugInformation` to match the event above. good catch!

**Review by nc6 — APPROVED** [review]

**Pull request merged**